tc/src/main/java/com/epam/tc/model/Course.java

```diff
@@ -20,6 +28,80 @@
     private String description;
     private String links;
 
+    @ManyToMany(cascade = CascadeType.ALL)
+    @JoinTable(name = "subscribers", joinColumns = {
+        @JoinColumn(name = "CourseId")}, inverseJoinColumns = {
+        @JoinColumn(name = "UserId")})
+    private Set<User> subscribers;
+
+    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL)
+    private Set<Evaluate> attenders;
+
+    @ManyToOne
+    @JoinColumn(name = "UserId")
+    private User owner;
+
+    public boolean isSubscribed(User user) {
+        return subscribers.contains(user);
+    }
+
+    public boolean isAttended(User user) {
+        return attenders.stream().anyMatch((attender) -> (attender.getUser() == user));
+    }
+
+    public boolean isHaveGrade(User user) {
+        return attenders.stream().anyMatch((attender)
+                -> ((attender.getUser() == user) && (attender.getGrade() != null)));
+    }
+
+    public double getAverageGrade() {
+        int ch = 0, sum = 0;
+        for (Evaluate attender : attenders) {
+            if (attender.getGrade() != null) {
+                sum += attender.getGrade();
+                ch++;
+            }
+        }
+        if (ch == 0) {
+            return 0;
+        } else {
+            return sum / ch;
+        }
+    }
+
+    public Set<User> getSubscribers() {
+        return this.subscribers;
+    }
+
+    public Set<Evaluate> getAttenders() {
+        return attenders;
+    }
+
+    public void setAttenders(Set<Evaluate> attenders) {
+        this.attenders = attenders;
+    }
+
+    public void addSubscriber(User subscriber) {
+        this.subscribers.add(subscriber);
+    }
+
+    public void deleteSubscriber(User subscriber) {
+        this.subscribers.remove(subscriber);
+    }
+
+    public void addAttender(User attender) {
+        Evaluate evaluate = new Evaluate(attender, this, null);
```

Comment:
If you have to pass null you'd consider rewrite your constructor or create another one with proper set of arguments.